aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -164,6 +164,32 @@ Tensor NestedTensor_gelu(const Tensor& self, c10::string_view approximate) {
       });
 }
 
+Tensor NestedTensor_nested_tensor_from_mask(const Tensor& t, const Tensor& mask, double min_padding_rate) {
+    TORCH_CHECK(mask.scalar_type() == at::ScalarType::Bool, "Mask tensor should be Bool");
+    TORCH_CHECK(mask.dim() == 2, "Padding mask should be 2D");
+    TORCH_CHECK(t.dim() == 3, "Input should be a 3D tensor, N * L * D");
+    // One assumption is mask is always padded at the end , never in the middle
```

Comment:
Can this assumption be validated efficiently for a nice error message if the mask is not structured correctly?

Reply:
I feel we can always validte it, but efficiently... Maybe no, it is always a O(mask.size()) solution?